django_emarsys/event.py

```diff
@@ -0,0 +1,263 @@
+        return num_new_events, num_updated_ids, num_deleted_ids, \
+            unknown_event_names, unsynced_event_names
+
+    for local_event in Event.objects.all():
+        emarsys_event_id = emarsys_events.get(local_event.name)
+        if emarsys_event_id:
+            if local_event.emarsys_id != emarsys_event_id:
+                num_updated_ids += 1
+
+            local_event.emarsys_id = emarsys_event_id
+            del emarsys_events[local_event.name]
+            local_event.save()
+        else:
+            local_event.delete()
+            num_deleted_ids += 1
+
+    for emarsys_event_name, emarsys_event_id in emarsys_events.items():
+        Event.objects.create(emarsys_id=emarsys_event_id,
+                             name=emarsys_event_name)
+        num_new_events += 1
+
+    all_synced_event_names = set(Event.objects.all()
+                                 .values_list('name', flat=True))
+    known_event_names = set(settings.EMARSYS_EVENTS.keys())
+    unknown_event_names = list(all_synced_event_names - known_event_names)
+    unsynced_event_names = list(known_event_names - all_synced_event_names)
+
+    if unknown_event_names:
+        log.warning("these Emarsys event names are not "
+                    "configured in settings.EMARSYS_EVENTS: {}"
+                    .format(', '.join(
+                        '"{}"'.format(x.replace('"', '\\"'))
+                            for x in unknown_event_names)))
+
+    if unsynced_event_names:
+        log.warning("these event names in settings.EMARSYS_EVENTS are not "
+                    "known by Emarsys: {}"
+                    .format(', '.join(
+                        '"{}"'.format(x.replace('"', '\\"'))
+                            for x in unsynced_event_names)))
+
+    return num_new_events, num_updated_ids, num_deleted_ids, \
+        unknown_event_names, unsynced_event_names
+
+
+def trigger_automatic_event(event_name, recipient_email, data=None,
+                            create_user_if_needed=True,
+                            contact_data_provider=None):
+    return _trigger_event(event_name=event_name,
+                          recipient_email=recipient_email,
+                          data=data,
+                          create_user_if_needed=create_user_if_needed,
+                          contact_data_provider=contact_data_provider,
+                          source=EventInstance.SOURCE_AUTOMATIC)
+
+
+def trigger_manual_event(event_name, recipient_email, data=None,
+                         create_user_if_needed=True,
+                         contact_data_provider=None):
+    return _trigger_event(event_name=event_name,
+                          recipient_email=recipient_email,
+                          data=data,
+                          create_user_if_needed=create_user_if_needed,
+                          contact_data_provider=contact_data_provider,
+                          source=EventInstance.SOURCE_MANUAL)
```

Comment:
I feel they belong to the model, because the model will store those values directly in the database. If the model changes in some way involving the `source` attribute, then all code triggering events breaks, or `event` has to introduce a mapping to handle the new situation, so it's a good idea to have that mapping right away.

I can think of three ways of doing this:
1. `trigger_automatic_event()` and `trigger_manual_event()`
2. `trigger_event(..., source="automatic")` and `trigger_event(..., source="manual")`
3. `trigger_event(..., source=event.AUTOMATIC)` and `trigger_event(..., source=event.MANUAL)` (or other constants in `__init__.py` that are *not* the same as in the model)

I'm least happy with 3., because of the extra import and knowledge necessary. I think 2. is sufficiently Pythonic, but of course has the typo factor. 1. is (in my eyes) best for the calling code and only has the disadvantage of those 2 extra functions with repeated argument lists.

That being said, I don't *mind* any of the solutions, so if you feel strongly in favour of either of those, then I'll happily implement that'un. :)